### ISSUE TITLE
Password protected toots

### ISSUE THREAD

**Issue by gaylie** [Taullo/mastodon#12]
### Pitch

A toot that the user will have to enter a password before viewing.

### Motivation

I love the privacy options on toots and i think this one would be a great step to make even more protected toots. It would allow the user to only share their toots with specific people. An alternative option I thought of for this was a permission list, but that would require the viewer to have an account. If the toot was protected by a password, you can give anyone permission to view it with or without aethy account, by giving them said password!

**Comment by BeebzVivian** [comment]
In lieu of a permission list, perhaps links with "permission keys"? akin to MEGA could be interesting. 

In addition, Use case could be if someone wanted to use aethy as a delivery method for fanbox / subscribestar / patreon rewards in lieu of something like telegram. The permission key/password method could help someone remain on a platform they feel comfortable with as opposed to managing multiple accounts.

**Comment by gaylie** [comment]
> In lieu of a permission list, perhaps links with "permission keys"? akin to MEGA could be interesting.
> 
> In addition, Use case could be if someone wanted to use aethy as a delivery method for fanbox / subscribestar / patreon rewards in lieu of something like telegram. The permission key/password method could help someone remain on a platform they feel comfortable with as opposed to managing multiple accounts.

Do by "links with permission keys" mean, like, a post that's usually hidden, unless you have the specific permission-link? If that's the case, it'd be nice to be able to later revoke permission for said links, in case they somehow get spread beyond the intended reach.

**Comment by BeebzVivian** [comment]
> Do by "links with permission keys" mean, like, a post that's usually hidden, unless you have the specific permission-link? If that's the case, it'd be nice to be able to later revoke permission for said links, in case they somehow get spread beyond the intended reach.

Yes so like for example with a google doc generating a link that can be seen if you have the link but you can always lock it back down later. I agree.

**Comment by Taullo** [comment]
I don't have any issues with this on principle but it'd require changes to the database in order to add the keys and I dunno what the implications would be for federation 🤔 if it was federated, the two ways I could see to do it would be:

- Giving access to the keys and the contents of the post to other Fediverse servers in order to display it locally which would defeat the point. Plus, they'd have to support the feature.
- Simply providing a link back to the origin server where users can then input the key/password. I feel like this is in opposition to how federation was designed though and would get pushback from other servers.

The posts could be local only but then it restricts their reach in general. Either way it'd require pretty substantial additions to the API in order to accomplish.

**Comment by gaylie** [comment]
I may not fully understand everything about federation yet, but I think in general the lack of reach in posts like that wouldn't necessarily an issue, at least that's how I imagined it! Like a local post, only available on aethy, but you don't need an account to view it, and only people with the password/key can view it.
The post being available on other Fediverse servers would likely just defeat the point of these types of posts.
